### DISCUSSION TITLE
Hosting packages of other languages?

### DISCUSSION THREAD

**Discussion by wongjiahau** [nestdotland/nest.land#312] [Q&A] [answered]
Is it possible to host packages that are not based on JavaScript/Typescript?

**Comment by SteelAlloy** [comment, marked as answer]
Nest.land is meant for javascript and typescript modules, most notably for Deno.
However nothing prevents you from publishing code for other languages, as long as it supports url imports. 🚀
In which language would you like to publish your code?

**Reply by wongjiahau** [reply]
Thanks, I would like to use it for my language, which I'm still developing currently.

**Reply by SteelAlloy** [reply]
Great, we look forward to having modules of your language on nest.land 🚀

**Reply by wongjiahau** [reply]
Looking at the current documentation, it seems like Deno is required for using the `egg` CLI, are there any alternative methods that can be used for publishing packages?

**Reply by SteelAlloy** [reply]
When creating nest.land, we didn't think that other languages could be used. However, we are currently in the process of rewriting the CLI for v1 (it will still be in Deno though, but it will show you how to publish a package from any language). We'll do our best to provide comprehensive documentation on the API.

**Reply by SteelAlloy** [reply]
@wongjiahau One more thing: Deno offers the possibility to compile scripts into a self-contained executable.
When this feature will be stable you will be able to use our CLI without installing Deno.

**Comment by littledivy** [comment]
Any language or tool that supports URL modules _can_ import from nest.land. Since url imports are a Web/Deno specific standard, nest.land gives them priority. What is your use case?

**Reply by wongjiahau** [reply]
I would like to use it for hosting packages for my language, and I'm particularly concern about reproducibility and security, and I think nest.land fits my requirements quite well.
